src/components/ui/flex.tsx

```diff
@@ -31,17 +41,46 @@ const Center = forwardRef<FlexElement, FlexProps>(
 
 Center.displayName = "Center"
 
-const Stack = forwardRef<FlexElement, FlexProps>(
-  ({ className, ...props }, ref) => {
+type StackProps = FlexProps & { separator?: ReactElement }
+
+const Stack = forwardRef<FlexElement, StackProps>(
+  ({ className, separator, children, ...props }, ref) => {
+    const cloneChildren = useMemo(() => {
```

Comment:
Applied approach similar to the stack component from Chakra to still allow for some separator component to be passed in and rendered internally for repetition.